src/webots/vrml/WbNode.cpp

```diff
@@ -1060,9 +1092,58 @@ QStringList WbNode::listTextureFiles() const {
   return list;
 }
 
+const QString WbNode::urdfName() const {
+  QString name;
+  if (this->findSFString("name")) {
```

Comment:
```suggestion
  if (isUrdfLinkRoot()) {
```

Reply:
This is not the same, `isUrdfLinkRoot()` can return true and it could happen that the node doesn't have field `name` defined. This happens when there is `joint-link-joint` and the link doesn't have a name. In that situation, we have to consider Webots node as URDF link even though there is no name.

Reply:
Oh yes, you're completely right!